motion_computation/test/MotionComputationTest.cpp

```diff
@@ -189,10 +192,101 @@ namespace motion_computation
+        input.path_history.initial_position.utc_time.year.year = 2000;
+        input.path_history.initial_position.utc_time.presence_vector |= j2735_v2x_msgs::msg::DDateTime::MONTH;
+        input.path_history.initial_position.utc_time.month.month = 10;
+        input.path_history.initial_position.utc_time.presence_vector |= j2735_v2x_msgs::msg::DDateTime::DAY;
+        input.path_history.initial_position.utc_time.day.day = 2;
+        input.path_history.initial_position.utc_time.presence_vector |= j2735_v2x_msgs::msg::DDateTime::HOUR;
+        input.path_history.initial_position.utc_time.hour.hour = 20;
+        input.path_history.initial_position.utc_time.presence_vector |= j2735_v2x_msgs::msg::DDateTime::MINUTE;
+        input.path_history.initial_position.utc_time.minute.minute = 20;
+        input.path_history.initial_position.utc_time.presence_vector |= j2735_v2x_msgs::msg::DDateTime::SECOND;
+        input.path_history.initial_position.utc_time.second.millisecond = 60998;
+
+        input.speed.velocity = 8.98;
+        // Set Confidence
+        input.accuracy.presence_vector |= carma_v2x_msgs::msg::PositionalAccuracy::ACCURACY_AVAILABLE;
+        input.accuracy.presence_vector |= carma_v2x_msgs::msg::PositionalAccuracy::ACCURACY_ORIENTATION_AVAILABLE;
+        input.accuracy.semi_major = 217;
+        input.accuracy.semi_minor = 217;
+        input.accuracy.orientation = 60050;
+
+        //Define position
+        input.position.latitude = 0;
+        input.position.longitude = 0;
+        input.position.elevation = 0;
+
+        input.heading.heading = 0;
+
+        input.presence_vector |= carma_v2x_msgs::msg::PSM::HAS_PATH_PREDICTION;
+        input.path_prediction.radius_of_curvature = 50.0;
+        
+        RCLCPP_INFO_STREAM(node->get_logger(), "Running conversion test with defined values");
+
+        std::string projection = "+proj=tmerc +lat_0=0.0 +lon_0=0.0 +k=1 +x_0=0 +y_0=0 +datum=WGS84 +units=m +vunits=m "
+                          "+no_defs";
+        std_msgs::msg::String msg;
+        msg.data = projection;
+        lanelet::projection::LocalFrameProjector local_projector(msg.data.c_str());
+
+        carma_perception_msgs::msg::ExternalObject output2;
+        std::unique_ptr<carma_v2x_msgs::msg::PSM> input_ptr2 = std::make_unique<carma_v2x_msgs::msg::PSM>(input);
+        conversion::convert(*input_ptr2, output2, map_frame, mcw.prediction_period_, 
+        mcw.prediction_time_step_, local_projector , mcw.ned_in_map_rotation_,   mcw.node_clock_);
+
+        //compare id
+        int id = 0;
+        for(int i = 0; i < input.id.id.size(); i++){
+            id |= input.id.id[i] << (8 * i);
+        }
+        ASSERT_EQ(id, output2.id);
+        ASSERT_EQ(output2.header.frame_id, map_frame);
+        //check if object size matches pedestrian
+        ASSERT_EQ(output2.size.x, 0.5);
+        ASSERT_EQ(output2.size.y, 0.5);
+        ASSERT_EQ(output2.size.z, 1.0);
+        
```

Comment:
The tests here need to be more fleshed out. At a minimum we need a test of the output position, heading, and timestamp since that will be the core acceptance criteria for CP.

Reply:
Unit testing is pretty useless if we don't actually validate the critical outputs